### PR TITLE
Exoplanet Overhaul: The Shape of Things to Come

### PR DESCRIPTION
**Pull request by JohnWildkins**
A complete overhaul of how exoplanet terrain is generated, loosely based off and inspired by tgstation/tgstation#51082.

Exoplanets now select from a series of biomes utilizing random noise maps, which in turn use various tuned generation systems to provide varied and distinct environments. That's a lot of buzzwords for saying that essentially, we go from having one random map theme (make some walls of minerals) to potentially infinite, all designed to be relatively easy to play with and modify.

Several new turfs / flora including new desert turfs and objects, as well as everything on the new fungal and crystal planets, by @kyres1 

Comparison shots:
### Asteroid
image
### Barren
image
### Desert
image
### Grass (currently unused)
image
### Grove/Jungle
image
### Lava
image
### Snow (sort of unused - Adhomai is special)
image

And two new exoplanet types, just for fun:
### Fungal Marsh
image
### Crystalline
image

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

**Comment by JohnWildkins** [comment]
oh yeah also this PR updates rust-g to v1.0.2+a4 for noise gen

File: code/__defines/rust_g.dm [hunk 1]
```diff
@@ -188,10 +213,10 @@
 	else
 		CRASH(output["content"])
 
-#define rustg_udp_send(addr, text) LIBCALL(RUST_G, "udp_send")(addr, text)
+#define rustg_udp_send(addr, text) call(RUST_G, "udp_send")(addr, text)
```
Comment (Arrow768) [inline code/__defines/rust_g.dm:216]: That looks like its going to cause issues with 515 (call vs ext_call vs RUSTG_CALL)

**Review by Arrow768 — APPROVED** [review]

File: code/__defines/rust_g.dm (hunk 1 above)
Comment (JohnWildkins) [reply code/__defines/rust_g.dm:216]: that's a problem with our rust-g fork then, because this is the autogenerated file from rust-g

**Comment by TheDanseMacabre** [comment]
roleplay is saved

**Pull request merged**